### PR TITLE
Ipywidgets 8 support

### LINKED ISSUE #366: JSON serialization of OrbitControls fails with jupyter_client 7.0.3
I believe a [recent change](https://github.com/jupyter/jupyter_client/commit/1363f53f308f0d3621950dcbd2e49edd89ab4aa6#diff-f89a7d6e5a8a1910d98e3a31ae8ff90c4d745d16bcef4a11d969e5459dcb7e50) in the json serialization within `jupyter_client` leads to the following error when trying to use `OrbitControls` in a jupyter notebook.

```python
ValueError: Out of range float values are not JSON compliant
```
The issue is caused by default values of `+/- Inf` for some parameters in `OrbitControls` 

This can also be seen in the second cell of the [built-in example](https://github.com/jupyter-widgets/pythreejs/blob/master/examples/Examples.ipynb).

Here is a minimal (non)-working example:

```python
import pythreejs
camera = pythreejs.PerspectiveCamera(
    position=[0.0, 0.0, 6.0],
    fov=20,
    aspect=1.35,
    near=0.001
)
controls = pythreejs.OrbitControls(
    controlling=camera, 
#     minAzimuthAngle=-1.0,
#     maxAzimuthAngle=1.0,
#     maxDistance=1.0,
#     maxZoom=1
)
```
Removing the commented lines will insert finite values for all parameters that would default `to +/- Inf` otherwise, after which the serialization works again.

System info:
```bash
python --version
Python 3.7.10

python -c "import pythreejs; print(pythreejs.version_info)"
(2, 3, 0, 'final')

 jupyter --version
Selected Jupyter core packages...
IPython          : 7.27.0
ipykernel        : 6.4.0
ipywidgets       : 7.6.5
jupyter_client   : 7.0.3
jupyter_core     : 4.8.1
jupyter_server   : not installed
jupyterlab       : not installed
nbclient         : 0.5.4
nbconvert        : 6.1.0
nbformat         : 5.1.3
notebook         : 6.4.4
qtconsole        : 5.1.1
traitlets        : 5.1.0
```

### PR DESCRIPTION
**Pull request by vidartf**
Ipywidgets 8 is above to be released, and some changes are needed to support it:

- [x] We cannot use the `.extend()` method from the base classes anymore. We should switch to using `class ... extends` pattern (switched to es2018 throughout).
- [x] We cannot use `_ipython_display_` for the preview anymore. We should switch to `_repr_mimebundle_`.
- [x] Add serializers for non-finite floats. Closes #366.
- [x] Dropped support for Python < 3.7, including Python 2.

### CODE REVIEW

File: js/src/_base/utils.js [hunk 1]
```diff
@@ -324,7 +324,7 @@ function nestedDiff(newObj, oldObj) {
     } else {
         diff = dictDiff(newObj, oldObj);
     }
-    var all = _.flatten([diff.added, diff.removed, diff.kept]);
+    var all = [...diff.added, ...diff.removed, ...diff.kept];
```
Comment (vidartf) [inline js/src/_base/utils.js:327]: This fixes a bug for listening to certain arrays in dicts structures.

**Comment by vidartf** [comment]
@akaszynski Do you want to do a review / test this?

**Comment by LtAirman** [comment]
Hello vidartf, 

I’m a newbie programmer working with Jupyter Notebook and pythreejs.
  
All 3 of my project files at https://github.com/LtAirman/ChargeFieldTopics stopped working this morning. They no longer display the pythreejs rendered scene. 

Over the last few months the cell setting up the rendered scene displayed a UserWarning, due to non JSON compliant infinite values. I believe you identified the problem, involving the pythreejs orbital camera, in issues 366 and 378 which I’ve been following.  

It was my impression this non-finite number problem would be corrected with the upcoming ipywidget 8 release, seeing no rendered output at all is an awful surprise. I assume it's the same issue. What do I do now?  

I’d appreciate any guidance or information you might provide in this matter.

**Comment by LtAirman** [comment]
Update, happy to report my projects' pythreejs rendered outputs are now working properly.

**Comment by LtAirman** [comment]
Along with the UserWarning.

**Pull request merged**